private/model/api/behavior.go

```diff
@@ -0,0 +1,320 @@
+				val += fmt.Sprintf("RequestBodyMatchesXML(t, %q, req, %v)", assertionContext, c.Request.EmptyShapeBuilder(&op.InputRef))
+			case "requestBodyEqualsString":
+				val += fmt.Sprintf("RequestBodyEqualsString(t, %q, req)", assertionContext)
+			case "requestIdEquals":
+				val += fmt.Sprintf("RequestIDEquals(t, %q, req.RequestID)", assertionContext)
+			case "responseDataEquals":
+				val += fmt.Sprintf("ResponseDataEquals(t, %v, resp)", c.BuildOutputShape(&op.OutputRef))
+			case "responseErrorDataEquals":
+				val += fmt.Sprintf("ResponseErrorDataEquals(t, %#v, err)", assertionContext)
+			case "responseErrorIsKindOf":
+				val += fmt.Sprintf("ResponseErrorIsKindOf(t, %q, err)", assertionContext)
+			case "responseErrorMessageEquals":
+				val += fmt.Sprintf("ResponseErrorMessageEquals(t, %q, err)", assertionContext)
+			case "responseErrorRequestIdEquals":
+				val += fmt.Sprintf("ResponseErrorRequestIDEquals(t, %q, err)", assertionContext)
+			default:
+				panic("Invalid assertion key")
+			}
+		}
+	}
+	return val
+}
+
+// Returns a value to set Credentials
+func (c Case) ConfigurationString(T Tests) string {
+	region := T.Defaults.Env["AWS_REGION"]
+	accessKeyId := T.Defaults.Env["AWS_ACCESS_KEY"]
+	secretAccessKey := T.Defaults.Env["AWS_SECRET_ACCESS_KEY"]
+	endpointUrl := ""
+	if len(c.LocalConfig) > 0 {
+		for key, val := range c.LocalConfig {
+			switch key {
+			case "region":
+				region = val
+			case "accessKeyId":
+				accessKeyId = val
+			case "secretAccessKey":
+				secretAccessKey = val
+			case "endpointUrl":
+				endpointUrl = val
+			}
+		}
+	} else {
+		for key, val := range c.LocalEnv {
+			switch key {
+			case "AWS_REGION":
+				region = val
+			case "AWS_ACCESS_KEY":
+				accessKeyId = val
+			case "AWS_SECRET_ACCESS_KEY":
+				secretAccessKey = val
+			case "EndPointURL":
+				endpointUrl = val
+			}
+		}
+	}
+	return fmt.Sprintf("\n\t\tRegion: aws.String(%#v),\n\t\tCredentials: credentials.NewStaticCredentials(%#v, %#v, %#v),",
+		region, accessKeyId, secretAccessKey, endpointUrl)
+}
+
+// ErrorAssertExists returns true if there is an assertion in the expect map
+// which requires error as an argument
+func (c Case) ErrorAssertExists() bool {
+	for _, assertion := range c.Expect {
+		for assertionName := range assertion {
+			if strings.Contains(assertionName, "Error") {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+//template map is defined in "eventstream.go"
+var funcMap = template.FuncMap{
+	"Map": templateMap,
+}
+
+var behaviorTestTmpl = template.Must(template.New(`behaviorTestTmpl`).Funcs(funcMap).Parse(`
+
+{{define "StashCredentials"}}
+	restoreEnv := sdktesting.StashEnv() //Stashes the current environment
+	defer restoreEnv()
+{{end}}
+
+{{define "SessionSetup"}}
+	//Starts a new session with credentials and region parsed from "defaults" in the Json file'
+	sess := session.Must(session.NewSession(&aws.Config{ {{$.testCase.ConfigurationString $.Tests}} }))
+{{end}}
+
+{{define "ResponseBuild"}}
```

Comment:
Left hand whitespace is a bit off, probably a mix of tabs and space characters. try to only use tabs.

Reply:
Also for sub template like this you can general be left aligned and don't have to worry about additional lefthand whitespace. gofmt will take care of this.